contracts/extensions/AmountGetterWithFee.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.23;
+
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+
+import { IOrderMixin } from "../interfaces/IOrderMixin.sol";
+import { AmountGetterBase } from "./AmountGetterBase.sol";
+
+/// @title Price getter contract that adds fee calculation
+contract AmountGetterWithFee is AmountGetterBase {
+    /// @dev Allows fees in range [1e-5, 0.65535]
+    uint256 internal constant _FEE_BASE = 1e5;
+    uint256 internal constant _DISCOUNT_BASE = 100;
+
+    /**
+     * @dev Calculates makingAmount with fee.
+     */
+    function _getMakingAmount(
+        IOrderMixin.Order calldata order,
+        bytes calldata extension,
+        bytes32 orderHash,
+        address taker,
+        uint256 takingAmount,
+        uint256 remainingMakingAmount,
+        bytes calldata extraData
+    ) internal view virtual override returns (uint256) {
+        unchecked {
+            (, uint256 integratorFee, uint256 resolverFee, bytes calldata tail) = _parseFeeData(extraData, taker, _isWhitelistedGetterImpl);
+            return Math.mulDiv(
```

Comment:
Was the removal of rounding `Math.Rounding.Floor` intentional?

Reply:
Yes, it is more efficient to use `mulDiv` without extra rounding argument as it is implies flooring by default. See [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/0df841d2d75352786532e9e26f400efdef78d86e/contracts/utils/math/Math.sol#L229).